helper-cli/src/main/kotlin/commands/GeneratePackageConfigurationsCommand.kt

```diff
@@ -0,0 +1,105 @@
+import com.github.ajalt.clikt.parameters.types.file
+
+import org.ossreviewtoolkit.helper.common.writeAsYaml
+import org.ossreviewtoolkit.model.Identifier
+import org.ossreviewtoolkit.model.OrtResult
+import org.ossreviewtoolkit.model.Provenance
+import org.ossreviewtoolkit.model.VcsType
+import org.ossreviewtoolkit.model.config.PackageConfiguration
+import org.ossreviewtoolkit.model.config.VcsMatcher
+import org.ossreviewtoolkit.model.readValue
+import org.ossreviewtoolkit.utils.expandTilde
+import org.ossreviewtoolkit.utils.safeMkdirs
+
+internal class GeneratePackageConfigurationsCommand : CliktCommand(
+    help = "Generates one package configuration for the source artifact scan and one for the VCS scan, if " +
+            "a corresponding scan result exists in the given ORT result for the respective provenance. The output " +
+            "package configuration YAML files are written to the given output directory."
+) {
+    private val ortResultFile by option(
+        "--ort-result-file",
+        help = "The input ORT file containing the package for which the package configurations shall be generated."
+    ).convert { it.expandTilde() }
+        .file(mustExist = true, canBeFile = true, canBeDir = false, mustBeWritable = false, mustBeReadable = false)
+        .convert { it.absoluteFile.normalize() }
+        .required()
+
+    private val packageId by option(
+        "--package-id",
+        help = "The target package for which the package configuration shall be generated."
+    ).convert { Identifier(it) }
+        .required()
+
+    private val outputDir by option(
+        "--output-dir",
+        help = "The output directory to write the package configurations to."
+    ).convert { it.expandTilde() }
+        .file(mustExist = false, canBeFile = false, canBeDir = true, mustBeWritable = false, mustBeReadable = false)
+        .convert { it.absoluteFile.normalize() }
+        .required()
+
+    override fun run() {
+        outputDir.safeMkdirs()
+
+        val ortResult = ortResultFile.readValue<OrtResult>()
+        val scanResults = ortResult.getScanResultsForId(packageId)
+
+        scanResults.find { it.provenance.vcsInfo != null }?.provenance
+            ?.writePackageConfigurationFile("vcs.yml")
+        scanResults.find { it.provenance.sourceArtifact != null }?.provenance
+            ?.writePackageConfigurationFile("source-artifact.yml")
+    }
+
+    private fun Provenance.writePackageConfigurationFile(filename: String) {
+        val packageConfiguration = createPackageConfiguration(packageId, this)
+        val outputFile = outputDir.resolve(filename)
+
+        packageConfiguration.writeAsYaml(outputFile)
+        println("Wrote a package configuration to '${outputFile.absolutePath}'.")
+    }
+}
+
+private fun createPackageConfiguration(id: Identifier, provenance: Provenance): PackageConfiguration =
+    if (provenance.vcsInfo != null) {
+        val vcsInfo = provenance.vcsInfo!!
```

Comment:
Ah, sorry, I didn't watch closely enough to understand that you liked the previous version better because it allow you to get rid of this cast. I agree we should avoid `!!` casts as much as possible as they're ugly and error prone. I'll still approve and leave it to you whether you again want to revert back or not.

Reply:
I reverted it back again.